package-system/python/docker_build_linux.sh

```diff
@@ -5,77 +5,158 @@
+
+
+# Prepare the dependent O3DE package information for SQLite
+SQLITE_BASE=$WORKSPACE/temp/${O3DE_SQLITE_PACKAGE}/SQLite
+echo "Using O3DE SQLite3 package from ${SQLITE_BASE}"
+
+
+# Prepare the dependent libffi package from github to use 
+LIBFFI_VERSION="v3.4.2"
+LIBFFI_GIT_URL="https://github.com/libffi/libffi.git"
+LIBFFI_SRC=ffi_src
+LIBFFI_SRC_PATH=${WORKSPACE}/${LIBFFI_SRC}
+LIBFFI_LIB_PATH=${WORKSPACE}/ffi_lib
+
+echo "Clone and build libFFI statically from ${FFI_GIT_URL} / ${LIBFFI_VERSION}"
+echo "git -C ${WORKSPACE} clone ${LIBFFI_GIT_URL} --branch ${LIBFFI_VERSION} --depth 1 ${LIBFFI_SRC}"
+git -C ${WORKSPACE} clone ${LIBFFI_GIT_URL} --branch ${LIBFFI_VERSION} --depth 1 ${LIBFFI_SRC}
+if [ $? -ne 0 ]
+then
+    echo "Failed cloning libffi from ${LIBFFI_GIT_URL}"
+    exit 1
+fi
+
+pushd ${LIBFFI_SRC_PATH}
+
+echo ./autogen.sh
+./autogen.sh
+if [ $? -ne 0 ]
+then
+    echo "'autogen' failed for libffi at ${LIBFFI_SRC_PATH}"
+    exit 1
+fi
+
+
+echo ./configure --prefix=${WORKSPACE}/${LIBFFI_LIB} --enable-shared=no CFLAGS='-fPIC' CPPFLAGS='-fPIC'
+./configure --prefix=$LIBFFI_LIB --enable-shared=no CFLAGS='-fPIC' CPPFLAGS='-fPIC' 
+if [ $? -ne 0 ]
+then
+    echo "'configure' failed for libffi at ${LIBFFI_SRC_PATH}"
+    exit 1
+fi
+
+echo make install
+make install
+if [ $? -ne 0 ]
+then
+    echo "'configure' failed for libffi at ${LIBFFI_SRC_PATH}"
+    exit 1
+fi
+
+popd
+
+# Build CPython from source
+
+echo "Building cpython from source ..."
 echo ""
 
-cd $WORKSPACE/cpython
+pushd ${SRC_PATH}
 
 # Build from the source with optimizations and shared libs enabled , and override the RPATH and bzip include/lib paths
-echo ./configure --prefix=$WORKSPACE/package/python --enable-optimizations --with-openssl=$WORKSPACE/openssl-local/build --enable-shared LDFLAGS='-Wl,-rpath=\$$ORIGIN:\$$ORIGIN/../lib:\$$ORIGIN/../.. -L../ffi_lib/lib' CPPFLAGS='-I../ffi_lib/include' CFLAGS='-I../ffi_lib/include' 
-./configure --prefix=$WORKSPACE/package/python --enable-optimizations --with-openssl=$WORKSPACE/openssl-local/build --enable-shared LDFLAGS='-Wl,-rpath=\$$ORIGIN:\$$ORIGIN/../lib:\$$ORIGIN/../.. -L../ffi_lib/lib' CPPFLAGS='-I../ffi_lib/include' CFLAGS='-I../ffi_lib/include' 
-retVal=$?
-if [ $retVal -ne 0 ]; then
-    echo "Error running configuring optimized build"
-    exit $retVal
+echo ./configure --prefix=${BUILD_FOLDER}/python --enable-optimizations --with-openssl=${OPENSSL_BASE} --enable-shared LDFLAGS='-Wl,-rpath=\$$ORIGIN:\$$ORIGIN/../lib:\$$ORIGIN/../.. -L../ffi_lib/lib -L'${SQLITE_BASE}'/lib' CPPFLAGS='-I../ffi_lib/include -I'${SQLITE_BASE}'' CFLAGS='-I../ffi_lib/include -I'${SQLITE_BASE}'' 
+./configure --prefix=${BUILD_FOLDER}/python --enable-optimizations --with-openssl=${OPENSSL_BASE} --enable-shared LDFLAGS='-Wl,-rpath=\$$ORIGIN:\$$ORIGIN/../lib:\$$ORIGIN/../.. -L../ffi_lib/lib -L'${SQLITE_BASE}'/lib' CPPFLAGS='-I../ffi_lib/include -I'${SQLITE_BASE}'' CFLAGS='-I../ffi_lib/include -I'${SQLITE_BASE}'' 
```

Comment:
repeat: the `set -o xtrace` option can be used to echo the command that executed.